sfge/src/main/java/com/salesforce/graph/symbols/apex/ApexForLoopValue.java

```diff
@@ -31,19 +26,23 @@ public final class ApexForLoopValue extends ApexPropertiesValue<ApexForLoopValue
         if (valueVertex != null
                 && !(valueVertex instanceof VariableExpressionVertex.ForLoop)
                 && !(valueVertex instanceof NewListInitExpressionVertex)
-                && !(valueVertex instanceof NewListLiteralExpressionVertex)) {
+                && !(valueVertex instanceof NewListLiteralExpressionVertex)
+                && !(valueVertex instanceof NewSetInitExpressionVertex)
+                && !(valueVertex instanceof NewSetLiteralExpressionVertex)) {
             throw new UnexpectedException(valueVertex);
         }
     }
 
-    ApexForLoopValue(ApexListValue value, ApexValueBuilder builder) {
+    ApexForLoopValue(ApexIterableCollectionValue value, ApexValueBuilder builder) {
```

Comment:
Most of the logic in this class remains the same. I've opened up the restraints to handle Sets the same way as Lists.